features/org.wso2.carbon.dashboards.portal.feature/src/main/resources/portal/extensions/widgets/Publisher/Publisher.js

```diff
@@ -0,0 +1,6 @@
){if(e)for(var n=0;n<e.length;n++)c(t,e[n])},childContextTypes:function(t,e){t.childContextTypes=i({},t.childContextTypes,e)},contextTypes:function(t,e){t.contextTypes=i({},t.contextTypes,e)},getDefaultProps:function(t,e){t.getDefaultProps?t.getDefaultProps=f(t.getDefaultProps,e):t.getDefaultProps=e},propTypes:function(t,e){t.propTypes=i({},t.propTypes,e)},statics:function(t,e){l(t,e)},autobind:function(){}},E={componentDidMount:function(){this.__isMounted=!0}},P={componentWillUnmount:function(){this.__isMounted=!1}},x={replaceState:function(t,e){this.updater.enqueueReplaceState(this,t,e)},isMounted:function(){return!!this.__isMounted}},_=function(){};return i(_.prototype,t.prototype,x),v}var i=n(2),u=n(10),a=n(0),s="mixins";t.exports=o},function(t,e,n){"use strict";function r(t){return i.isValidElement(t)||o("143"),t}var o=n(3),i=n(1);n(0);t.exports=r},function(t,e){var n;n=function(){return this}();try{n=n||Function("return this")()||(0,eval)("this")}catch(t){"object"==typeof window&&(n=window)}t.exports=n},function(t,e,n){"use strict";function r(t,e){if(!(t instanceof e))throw new TypeError("Cannot call a class as a function")}function o(t,e){if(!t)throw new ReferenceError("this hasn't been initialised - super() hasn't been called");return!e||"object"!=typeof e&&"function"!=typeof e?t:e}function i(t,e){if("function"!=typeof e&&null!==e)throw new TypeError("Super expression must either be null or a function, not "+typeof e);t.prototype=Object.create(e&&e.prototype,{constructor:{value:t,enumerable:!1,writable:!0,configurable:!0}}),e&&(Object.setPrototypeOf?Object.setPrototypeOf(t,e):t.__proto__=e)}Object.defineProperty(e,"__esModule",{value:!0});var u=function(){function t(t,e){for(var n=0;n<e.length;n++){var r=e[n];r.enumerable=r.enumerable||!1,r.configurable=!0,"value"in r&&(r.writable=!0),Object.defineProperty(t,r.key,r)}}return function(e,n,r){return n&&t(e.prototype,n),r&&t(e,r),e}}(),a=n(6),s=function(t){return t&&t.__esModule?t:{default:t}}(a),c=function(t){function e(t){r(this,e);var n=o(this,(e.__proto__||Object.getPrototypeOf(e)).call(this,t));return n.getDashboardAPI=n.getDashboardAPI.bind(n),n.subscribe=n.subscribe.bind(n),n}return i(e,t),u(e,[{key:"subscribe",value:function(t){this.props.glEventHub.on(this.props.id,t)}},{key:"publish",value:function(t){var e=this.props.id;this.props.glEventHub.emit(e,t)}},{key:"getDashboardAPI",value:function(){function t(){return""===window.location.hash||"#"===window.location.hash?{}:JSON.parse(window.location.hash.substr(1))}function e(t){window.location.hash=JSON.stringify(t)}function n(){var e=t();return e.hasOwnProperty(o.props.id)?e[o.props.id]:{}}function r(n){var r=t();r[o.props.id]=n,e(r)}var o=this;return{state:{get:function(t){var e=n();return e.hasOwnProperty(t)?e[t]:null},set:function(t,e){var o=n();o[t]=e,r(o)}}}}},{key:"render",value:function(){var t={padding:"30px 15px 15px 15px"};return s.default.createElement("div",{style:t},this.renderWidget())}}]),e}(a.Component);e.default=c}]);
```

Comment:
What does this file do? And why is it minified ?

Reply:
This is added to portal app until we properly handle the sample as discussed offline.